agent/__init__.py

```diff
@@ -542,14 +542,16 @@ def clear_credentials(path):
                 credentials_grouped[owner][name] = {}
             credentials_grouped[owner][name][cred['key']] = cred['value']
 
+        root_pw = pwd.getpwnam("root")
+
         for owner in credentials_grouped:
 
-            pw = pwd.getpwnam("root")
+            pw = root_pw  # if no owner, use 'root'
             if owner:
                 try:
                     pw = pwd.getpwnam(owner)
-                except KeyError:
-                    pw = pwd.getpwnam("root")
+                except KeyError:  # if invalid owner, use 'root'
+                    pw = root_pw
```

Comment:
`pass`

Reply:
may be so.
I assumed a slightly different behavior. But I tend to agree with you.

Reply:
changed

Reply:
So now we just skip a credential if its owner doesn't exist?

Reply:
yes. exactly.